bfss3/bfss3.go

```diff
@@ -397,8 +403,35 @@ func (i *iterator) fetchNextPage() error {
 		if ok, err := doublestar.Match(i.pattern, name); err != nil {
 			return err
 		} else if ok {
-			i.page = append(i.page, name)
+			i.page = append(i.page, object{
+				key:  name,
+				meta: newMetaInfoFromObject(name, obj),
```

Comment:
Hmmm, now I start to question how useful this feature really is. The discrepancy between implementations is the main reason why I didn't add a `Meta()` it in the first place. While e.g. the GS client allows you to retrieve full metadata from the iterator, the S3 implementation only provides a partial and requires you to issue an extra `Head` call to obtain the remaining details. 

Maybe it's more appropriate to expose the `Size()` and `ModTime()` as extra methods on the iterator instead to avoid this ambiguity?

Reply:
In [the declaration](https://github.com/bsm/bfs/blob/a9e3926/bfs.go#L69-L70) for `MetaInfo`, the remaining fields are marked optional:

```go
type MetaInfo struct {
	Name        string            // base name of the object
	Size        int64             // length of the content in bytes
	ModTime     time.Time         // modification time
	ContentType string            // optional content type
	Metadata    map[string]string // optional metadata
}
```

I figured there would be no harm in omitting optional fields. However, I agree that `Iterator` should grow specific metadata accessors if all fields on `MetaInfo` must be treated as mandatory.

Reply:
@dim Gentle bump: please help me make the decision here. To recap, there appear to be two options on the table:

1. Leave the PR as is because the fields `ContentType` and `Metadata` on `MetaInfo` really are optional and can be left unpopulated for that reason.
2. Remove the "optional" label from those fields, and update this PR to add `Size()` and `ModTime()` as extra methods on the iterator.

I don't have a strong preference for either option, but I think option 2 is cleaner and more explicit.

Reply:
Agree, would you be able to adapt the PR for option 2? Thanks, D

Reply:
Done!